### DISCUSSION TITLE
Nginx Proxy Manager - Update error

### DISCUSSION THREAD

**Discussion by di5cord20** [tteck/Proxmox#1993] [Help Needed] [answered]
I got the following error when trying to update the container.  Any thoughts on what's causing the error or how to fix?

`[ERROR] in line 127: exit code 3: while executing command wget -q "https://github.com/just-containers/s6-overlay/releases/download/v3.1.5.0/s6-overlay-noarch.tar.xz"`

**Comment by tteck** [comment]
No idea, exit code 3 suggests that there was a problem with the `wget` command execution, but [the URL](https://github.com/just-containers/s6-overlay/releases/download/v3.1.5.0/s6-overlay-noarch.tar.xz) is valid.

**Reply by di5cord20** [reply]
I restored a backup to try again.  As per the first try, I got the following error.  I forgot about that error and only posted the subsequent error on the 2nd try (which again shows for me now on the restored version).  Perhaps this new information can point to a root cause??

`[ERROR] in line 137: exit code 1: while executing command yarn install --network-timeout=30000 &> /dev/null`

**Reply by di5cord20** [reply, marked as answer]
I think I figured out the issue.  After another restore, I noticed the Bootsize disk was running low (had 4G storage).  I did a clean restore from my backup, upped storage to 6G, then ran the update script.  This time it completed successfully.

@tteck , not sure if it's possible, perhaps a check on free space before running the script to ensure sufficient space available?

**Reply by tteck** [reply]
I'll look into that. If you utilize the "Proxmox VE LXC Updater" it shows disk information.

**Reply by Stephan-4711** [reply]
Mine shows 79% full and I have the same issue after update  the contianer.
systemctl status nginx gives me a:

x nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; preset: enabled)
     Active: failed (Result: timeout) since Mon 2023-11-06 09:40:25 CET; 32s ago
       Docs: man:nginx(8)
    Process: 131 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
    Process: 184 ExecStart=/usr/sbin/nginx -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
        CPU: 311ms

Nov 06 09:38:53 nginxproxymanager systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Nov 06 09:38:55 nginxproxymanager systemd[1]: nginx.service: Can't open PID file /run/nginx.pid (yet?) after start: No such file or directory
Nov 06 09:40:25 nginxproxymanager systemd[1]: nginx.service: start operation timed out. Terminating.
Nov 06 09:40:25 nginxproxymanager systemd[1]: nginx.service: Killing process 198 (nginx) with signal SIGKILL.
Nov 06 09:40:25 nginxproxymanager systemd[1]: nginx.service: Killing process 199 (nginx) with signal SIGKILL.
Nov 06 09:40:25 nginxproxymanager systemd[1]: nginx.service: Killing process 200 (nginx) with signal SIGKILL.
Nov 06 09:40:25 nginxproxymanager systemd[1]: nginx.service: Killing process 201 (nginx) with signal SIGKILL.
Nov 06 09:40:25 nginxproxymanager systemd[1]: nginx.service: Failed with result 'timeout'.
Nov 06 09:40:25 nginxproxymanager systemd[1]: Failed to start nginx.service - A high performance web server and a reverse proxy server.y

it's the second time after updating the container, that it is broken after the update.
During starting the service, nginx is shortly working an then crashes.

**Reply by Stephan-4711** [reply]
after restore of the old verison systemctl status still shows it's not working. but the nginx services started. looks like it's trying to start twice?

nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Mon 2023-11-06 10:12:47 CET; 2min 52s ago
       Docs: man:nginx(8)
    Process: 155 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
    Process: 169 ExecStart=/usr/sbin/nginx -g daemon on; master_process on; (code=exited, status=1/FAILURE)
        CPU: 273ms

Nov 06 10:12:46 nginxproxymanager nginx[169]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Nov 06 10:12:46 nginxproxymanager nginx[169]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Nov 06 10:12:46 nginxproxymanager nginx[169]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
Nov 06 10:12:46 nginxproxymanager nginx[169]: nginx: [emerg] bind() to [::]:443 failed (98: Address already in use)
Nov 06 10:12:46 nginxproxymanager nginx[169]: nginx: [emerg] bind() to 0.0.0.0:81 failed (98: Address already in use)
Nov 06 10:12:46 nginxproxymanager nginx[169]: nginx: [emerg] bind() to [::]:81 failed (98: Address already in use)
Nov 06 10:12:47 nginxproxymanager nginx[169]: nginx: [emerg] still could not bind()
Nov 06 10:12:47 nginxproxymanager systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
Nov 06 10:12:47 nginxproxymanager systemd[1]: nginx.service: Failed with result 'exit-code'.
Nov 06 10:12:47 nginxproxymanager systemd[1]: Failed to start nginx.service - A high performance web server and a reverse proxy server.

**Reply by tteck** [reply]
`Address already in use`?

**Reply by Stephan-4711** [reply]
Yea this is the message. But at the moment it's running. When I restart, I get the nginx.service: Can't open PID file /run/nginx.pid  message. Very weird in my eyes. I have no Idea how to solve this problem.
I just used your script to deploy the nginx server. only installed joe and used the web if to set the reverse proxy settings. no clue what could cause this weirdo behaviour.

**Reply by tteck** [reply]
Does the output from `systemctl list-units --type=service` show a `npm.service` and a `nginx.service` file?
The script creates a `/lib/systemd/system/npm.service` where did the `/lib/systemd/system/nginx.service` come from?

**Reply by Stephan-4711** [reply]
` UNIT                               LOAD   ACTIVE SUB     DESCRIPTION                                             
  console-getty.service              loaded active running Console Getty
  container-getty@1.service          loaded active running Container Getty on /dev/tty1
  container-getty@2.service          loaded active running Container Getty on /dev/tty2
  cron.service                       loaded active running Regular background program processing daemon
  dbus.service                       loaded active running D-Bus System Message Bus
  ifupdown-pre.service               loaded active exited  Helper to synchronize boot up for ifupdown
  ifupdown-wait-online.service       loaded active exited  Wait for network to be configured by ifupdown
  networking.service                 loaded active exited  Raise network interfaces
  nftables.service                   loaded active exited  nftables
* nginx.service                      loaded failed failed  A high performance web server and a reverse proxy server
  npm.service                        loaded active running Nginx Proxy Manager
  openresty.service                  loaded active running The OpenResty Application Platform
  postfix.service                    loaded active exited  Postfix Mail Transport Agent
  postfix@-.service                  loaded active running Postfix Mail Transport Agent (instance -)
  ssh.service                        loaded active running OpenBSD Secure Shell server
  systemd-journal-flush.service      loaded active exited  Flush Journal to Persistent Storage
  systemd-journald.service           loaded active running Journal Service
  systemd-logind.service             loaded active running User Login Management
  systemd-modules-load.service       loaded active exited  Load Kernel Modules
  systemd-network-generator.service  loaded active exited  Generate network units from Kernel command line
  systemd-networkd.service           loaded active running Network Configuration
  systemd-remount-fs.service         loaded active exited  Remount Root and Kernel File Systems
  systemd-sysctl.service             loaded active exited  Apply Kernel Variables
  systemd-sysusers.service           loaded active exited  Create System Users
  systemd-tmpfiles-setup-dev.service loaded active exited  Create Static Device Nodes in /dev
  systemd-tmpfiles-setup.service     loaded active exited  Create Volatile Files and Directories
  systemd-update-utmp.service        loaded active exited  Record System Boot/Shutdown in UTMP
  systemd-user-sessions.service      loaded active exited  Permit User Sessions`

I have no idea where it's comming from. I just used your script. you see the failed nginx.service? nginx ist running right now

**Reply by tteck** [reply]
Try a `systemctl disable --now nginx.service` to see if that fixes this mishap.

**Reply by tteck** [reply]
`nginx.service` did not come from the script.

**Reply by Stephan-4711** [reply]
weird, seems that was the cause.
Now update went well. I have no Idea, why there is an nginx.service.
Maybe I'll backup the config manually and reinstall the container.

**Reply by Stephan-4711** [reply]
@tteck thanks for your help. I my self wouldn't have found that problem
